core/dbt/include/global_project/macros/materializations/common/merge.sql

```diff
@@ -87,6 +90,9 @@
 {% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
     {%- set predicates = [] if predicates is none else [] + predicates -%}
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+    {%- set sql_header = config.get('sql_header', none) -%}
+
+    {{ sql_header if sql_header is not none }}
```

Comment:
I think, if we do this, then the `sql_header` will run twice for "dynamic" (scripting) `insert_overwrite` materializations. (The first time will be when creating the temp table.) I assume that's ok?

Reply:
This is a good call. Temporary UDFs can't be defined twice in the same BQ script. Think we're going to need to find an alternative approach here.

I kicked around the idea of making this a materialization-level thing, and while I think that might be a good approach long-term, I don't want to touch every dbt materialization in support of this change. Going to try to figure out a nice way to make this sensible for a very specific use case (insert_overwrite on BigQuery incremental models) and localize the jank as much as possible